### DISCUSSION TITLE
Improving YOLO NAS inference time

### DISCUSSION THREAD

**Discussion by Idoleshem** [mikel-brostrom/yolo_tracking#1097] [Q&A]
Hi Mikel, amazing repo!
I would appreciate your help with a quick question :)

I tried to test YOLO_NAS_S + STRONGSORT performance and the inference time was much higher than what Deci had reported. 
My results were around ±120 ms per image, while yolov8n was around 15ms.

Do you any idea how can I address this latency issue and improve the performance?
I turned off all the parameters which responsible for saving images and the cropped_id, but no measure improvement was observed.

Thanks!

**Comment by mikel-brostrom** [comment]
It seems like there's a lot of overhead in the model.predict() function... https://github.com/Deci-AI/super-gradients/issues/958

**Comment by mikel-brostrom** [comment]
I manged to improve the yolo_nas inference x2. https://github.com/mikel-brostrom/yolo_tracking/pull/1098. Should be running at ~60ms per image. But the overhead in the computations for `.pt` inference is still quite big and it is a known issue.

**Reply by IdoLeshem1** [reply]
Thanks! I appreciate that :)

**Comment by goldmyu** [comment]
Any news regarding that issue @mikel-brostrom ? it really hinders the performance that Deci are promising with Yolo-NAS

**Reply by mikel-brostrom** [reply]
This issue is still there. https://github.com/Deci-AI/super-gradients/issues/958#issuecomment-1700432971. No answers

**Comment by Sumeshbaba** [comment]
Hello Mikel! Any update regarding this issue? The speed still seems to be quite low on PyTorch.

**Reply by mikel-brostrom** [reply]
Nope :cry:. It does not seem to be a priority for Deci to fix this

**Reply by Sumeshbaba** [reply]
This seems quite counterintuitive! When YOLOv7 and v8 can perform at about 15-20 ms, YOLO-NAS taking about 60-80 ms on the same device is not ideal! Hope they work on it soon!
